app/scripts/lib/createDupeReqFilterMiddleware.ts

```diff
@@ -0,0 +1,85 @@
+import { JsonRpcMiddleware } from 'json-rpc-engine';
+import log from 'loglevel';
+import { MINUTE } from '../../../shared/constants/time';
+
+export const THREE_MINUTES = MINUTE * 3;
+
+/**
+ * Creates a set abstraction whose values expire after three minutes.
+ *
+ * @returns The expiry set.
+ */
+const makeExpirySet = () => {
+  const seenValues: (number | string)[] = [];
+  const timestamps: number[] = [];
```

Comment:
any reason we shouldn't use a map instead? would make lookup easier too

Reply:
I went with the dual arrays because I was optimizing for deletion, but, on reflection, we ought to optimize for lookup, and can probably eat the performance costs when it comes to deletion.

I've replaced the dual arrays with a map.